yjit/src/core.rs

```diff
@@ -396,9 +396,13 @@ pub struct IseqPayload
 }
 
 impl IseqPayload {
-    // Empty the version map on the payload and return it
-    pub fn take_version_map(&mut self) -> VersionMap {
-        mem::take(&mut self.version_map)
+    /// Remove all block versions from the payload and then return them as an iterator
+    pub fn clear_all_blocks(&mut self) -> impl Iterator<Item = BlockRef> {
```

Comment:
Would `take_all_blocks` be a more Rusty naming here? I wouldn't expect something named `clear_` to return anything useful, I don't think.